json_encoder.go

```diff
@@ -129,6 +129,18 @@ func (enc *jsonEncoder) AddInt64(key string, val int64) {
 	enc.bytes = strconv.AppendInt(enc.bytes, val, 10)
 }
 
+// AddUInt adds a string key and integer value to the encoder's fields. The key
+// is JSON-escaped.
+func (enc *jsonEncoder) AddUInt(key string, val uint) {
+	enc.AddInt64(key, int64(val))
```

Comment:
we'll need to use `strconv.AppendUint` here, otherwise very large values will actually be printed as negative numbers (since the `int64` will overflow).

Same below for `AddUint64`. You probably want to use `strconv.AppendUint` int `AddUint64`, and `AddUint` can just call `AddUint64`.

@akshayjshah We should consider simplifying the Encoder interface for `int64` and `uint64` to remove `AddInt` and `AddUint` and just have the 64-bit versions, and the field can cast itself.